File: 9/list-9.go
package main

import "fmt"

type Imprimivel interface {
	Imprimir() string
}

type Relatorio struct{}

func (r Relatorio) Imprimir() string {
	return "Imprimindo o relatório..."
}

type Contrato struct{}

func (c Contrato) Imprimir() string {
	return "Imprimindo o contrato..."
}

func processarImpressao(imprimivel Imprimivel) {
	fmt.Println(imprimivel.Imprimir())
}

func main() {
	relatorio := Relatorio{}
	contrato := Contrato{}

	processarImpressao(relatorio)
	processarImpressao(contrato)
}

File: 9/list-9.py
from abc import ABC, abstractmethod

class Imprimivel(ABC):
    @abstractmethod
    def imprimir(self) -> str:
        pass

class Relatorio(Imprimivel):
    def imprimir(self) -> str:
        return "Imprimindo o relatório..."

class Contrato(Imprimivel):
    def imprimir(self) -> str:
        return "Imprimindo o contrato..."

def processar_impressao(imprimivel: Imprimivel):
    print(imprimivel.imprimir())

def main():
    relatorio = Relatorio()
    contrato = Contrato()

    processar_impressao(relatorio)  
    processar_impressao(contrato)

if __name__ == "__main__":
    main()